contracts/tests/test_validator_set.py

```diff
@@ -142,6 +142,49 @@ def test_cannot_call_finalize_change(validator_set_contract_session, accounts):
         contract.functions.finalizeChange().transact({"from": accounts[0]})
 
 
+def test_finalize_change_update_history_data_correctly(
```

Comment:
Probably better to split this up into multiple tests, each testing one state transition.

Reply:
My proposal is the following split:
- check that finalization is necessary (noted by you [here](https://github.com/trustlines-protocol/blockchain/pull/74#discussion_r282150530))
- check if new epoch start height has been added to history
- check if new validator set has been added to history

Enough or even more cases?

Reply:
Test code has been simplified much by the new interface.